src/lib/pages/workflows-with-new-search.svelte

```diff
@@ -136,20 +135,21 @@
     const { namespace } = $page.params;
     const { reason } = event.detail;
 
-    if (allSelected) {
-      // Idea: persist the job ID and display a progress indicator for large jobs
-      await batchTerminateByQuery({
-        namespace,
-        reason,
-        query: batchOperationQuery,
-      });
-      toaster.push({
-        message: 'The batch terminate request is processing in the background.',
-        id: 'batch-terminate-success-toast',
-      });
-    } else {
-      terminating = true;
-      try {
+    try {
```

Comment:
There might be an opportunity to reuse some of this logic in a reusable function e.g.

```js
  const batchAction = async (
    event: CustomEvent<{ reason: string }>,
    action: 'cancel' | 'terminate',
  ) => {
    const { namespace } = $page.params;
    const { reason } = event.detail;

    try {
      if (allSelected) {
        if (action === 'cancel') {
          await batchCancelByQuery({
            namespace,
            reason,
            query: batchOperationQuery,
          });
        }
        if (action === 'terminate') {
          await batchTerminateByQuery({
            namespace,
            reason,
            query: batchOperationQuery,
          });
        }
        toaster.push({
          message: `The batch ${action} request is processing in the background.`,
          id: `batch-${action}-success-toast`,
        });
      } else {
        const jobId = await bulkCancelByIDs({
          namespace,
          reason,
          workflows: cancelableWorkflows,
        });
        const workflows = await pollBatchOperation({
          namespace,
          jobId,
        });
        const COMPLETED_ACTION = {
          cancel: 'cancelled',
          terminate: 'terminated',
        };
        toaster.push({
          message: `Successfully ${COMPLETED_ACTION[action]} ${workflows} workflows.`,
          id: `batch-${action}-success-toast`,
        });
      }
    } catch (error) {
      toaster.push({
        message: `Unable to ${action} Workflows.`,
        variant: 'error',
      });
    } finally {
      resetPageToDefaultState();
    }
  };
```

But maybe something a little cleaner than that 😅